### DISCUSSION TITLE
Unable to synchronize with apache tinkerpop server

### DISCUSSION THREAD

**Discussion by anjalpetty** [aws/graph-explorer#456] [Q&A]
## Description

I have Tinkerpop server running at port 8182 on my local (Mac M1) and I've no problem connecting, creating graphs through python code. But I'm unable to connect with Graph explorer with the following connection url 

GraphType - Gremlin PG (Property Graph)
https://localhost:443

## Environment

- OS: Mac M1
- Browser: Safari
- Graph Explorer Version: 1.7.0
- Graph Database & Version: Apache Tinkerpop 3.7.2

## Steps to Reproduce

- Download apache-tinker-pop-gremlin-serve-3.7.2
- set the following properties in `conf/tinkergraph-empty.properties`

  ```
  gremlin.graph=org.apache.tinkerpop.gremlin.tinkergraph.structure.TinkerGraph
  gremlin.tinkergraph.vertexIdManager=ANY
  gremlin.tinkergraph.edgeIdManager=ANY
  ```

- run 
  ```
  $ bin/gremlin-server.sh start
  ```

- Running graph explorer as
  ```
  docker pull public.ecr.aws/neptune/graph-explorer:latest
  docker run -dit -p 443:443 --platform linux/amd64 --name mygraphexplorer --env HOST=localhost public.ecr.aws/neptune/graph-explorer:latest
  ```

![image](https://github.com/aws/graph-explorer/assets/73304513/5dee2649-547f-45b0-9cf8-31e3d7cf9c65)
#### Something went wrong
An error occurred. Please try again.

![image](https://github.com/aws/graph-explorer/assets/73304513/5dac98b6-c795-456f-b85b-6557e89336ae)

---

> [!IMPORTANT]
> If you are interested in working on this issue or have submitted
> a pull request, please leave a comment.

> [!TIP]
> Please use a 👍 reaction to provide a +1/vote.
>
> This helps the community and maintainers prioritize this request.

**Comment by kmcginnes** [comment]
Hi @anjalpetty. I'm sorry you are having issues.

Graph Explorer connects to the server through HTTP, which must be enabled on Gremlin Server. You can do so in the configuration file, or use the provided `gremlin-server-rest-modern.yaml ` config file.

You can find more information in the following places:

- [Connecting To Gremlin Server](https://github.com/aws/graph-explorer/blob/main/additionaldocs/connections.md#connecting-to-gremlin-server)
- [Local Gremlin Server Database](https://github.com/aws/graph-explorer/tree/main/additionaldocs/getting-started#gremlin-server-database)

I also noticed that in the image you provided the database URL is listed as 

```
https://localhost:443
```

But the server you mentioned was on port 8182 (the default port). So you might need to update the URL in the Graph Explorer connection to be:

```
https://localhost:8182
```

**Comment by kmcginnes** [comment]
@anjalpetty I'm going to convert this issue to a discussion, since it seems likely the issue is one of configuration.

Please let me know if my tips above helped at all. If not, please provide a bit more information about your setup.
